example/CMakeLists.txt

```diff
@@ -37,6 +37,9 @@ add_executable(coroutine_echo_client
 )
 add_dependencies(coroutine_echo_client crux)
 target_link_libraries(coroutine_echo_client crux ${EXTRA_LIBS})
+if(MSVC)
+  set_target_properties(coroutine_echo_client PROPERTIES LINK_FLAGS "${LINK_FLAGS} /SAFESEH:NO")
```

Comment:
I think that would be a fairly meaty task.  I know of other issues in this script, not just on Windows; e.g where multiple different versions of Boost are installed on a single machine (mismatch between the included headers and linked lib version) or static libs requested but dynamic libs linked.

It would be very sporting of Boost to just provide a CMake [config-file package](http://www.cmake.org/cmake/help/v3.1/manual/cmake-packages.7.html#config-file-packages) :-)